code/modules/projectiles/gun.dm

```diff
@@ -274,6 +275,12 @@
 
 	add_fingerprint(user)
 
+	// Handles smartgun alert level restriction.
+	if(smartgun && get_security_level() == "green" && firemodes[sel_mode].name != "stun") // Prevents smartguns from firing if they're not sent to stun
```

Comment:
1. A stun firemode won't always be called "stun", this has to be more generalized
2. Use the SEC_LEVEL_GREEN define

Reply:
1. This was the hardest part of this PR not even joking. I cannot get this code to work in any other way than this specific configuration. Please please please for the love of god if you can find a better way to do this be my guest, I've been working on that one specific line of code for 8 hours. PLEASE.
2. I tried to use SEC_LEVEL_GREEN but it doesn't work for some reason.